### DISCUSSION TITLE
block list of ip-addresses

### DISCUSSION THREAD

**Discussion by averlon** [fail2ban/fail2ban#3554] [Q&A] [answered]
Hi,
I am new to fail2ban.
I have setup a jail:
```
[ip-blacklist]
enabled = true
usedns = no
port     = anyport
action   = route
filter   = gwfwsfi_ip_blacklist
logpath  = /etc/fail2ban/gwfwsfi_ip_blacklist
maxretry = 0
findtime = 15552000
bantime  = -1
```
My logpath is a simple file which contains CIDR to block or ignore.
```
-109.207.200.47
-1.233.207.0/24
-1.80.0.0/13
+192.168.100.0/24
```

My filter looks like this but it seems not to work:
```

prefregex = ^(\-|\+)?<SUBNET>

failregex = ^\-<SUBNET>

datepattern = {NONE}
```

banned after reload:
```
fail2ban-client banned
[{'ip-blacklist': []}]
```

I would apreciate any help what could be wrong with the filter.
Thanks

**Comment by averlon** [comment]
I reloaded again, and now the blocked IPs are shown - I don't know why now - but anyhow, it seems to work now.

**Comment by sebres** [comment, marked as answer]
I have no idea why it did not work (to few info, e. g. log from fail2ban, etc), excepting probably the fact that if the jail and its logpath already known (for instance you already had that jail but incorrect failregex), fail2ban will try to recognize the rotation of file (check whether first line is changed) and if it is not new (not rotated), it'd seek to last known position (to find new lines only). This is more harder in your case, because there is no timestamp in log (so if you'd rewrite the file, first line can remain the same).

However there are some issues with the jail:
- you don't need `prefregex` here (it is for prefiltering purposes only if one has several `failregex` and some common part in different messages)... in your case it'd just unneeded waste CPU resources.
- `maxretry = 0` is incorrect (you have to change it to `maxretry = 1`)...
- `findtime = 15552000` (180d) is too large, but it doesn't matter size your `maxretry` is 0 (or 1), so it goes banned immediately by first attempt, and after restart fail2ban cannot seek 180 days in log back, because there are no timestamps.

By the way, you don't need the filter (`failregex` and `datepattern` can be specified directly in jail).

As for persistent banning (`bantime = -1`), it is not recommended and has many cons:
- never purging permanent bans unnecessary grow the list of banned IPs to up to immense amount;
- old entries will waste system resources, and the longer the more, in particular will bother net subsystem (routing or net filtering) and produce extra system load, etc;
- and everything that totally unneeded, because spend time for processing of IPs/nets that will never come back, especially if one consider than evildoers often change the IPs, let alone dynamic IPs;
- preventively preclude possible legitimate hosts or subnets (again dynamic or changed IPs) from the date they becomes rightful.

Even with incremental banning (`bantime.increment = true`), it is recommended to set `bantime.maxtime` to several weeks, e. g. `5w` is fully enough - if some recidive evildoer returns back after 5 weeks ban, it will be immediately banned again for 5 weeks (so basically it'd get only 10 attempts in year), but other banned occasionally so not a repeat offenders become forgotten.

I understand that your blacklist file is something artificial (written by you?), but if the hosts or subnets after repeated attempts becomes inserted again later, it'd be better to set some shorter `bantime` (and probably use `bantime.increment = true`).

**Comment by averlon** [comment]
I do have an additional question:
What does this message in the logfile tell me:
```
 2023-08-16 09:53:26,969 fail2ban.filter         [2710]: INFO    [ip-blacklist] Found 104.236.0.0/16 - 2023-08-16 09:53:26
```
This IP was banned by fail2ban jail "ip-blacklist"?

**Reply by sebres** [reply]
No, as the message says, it was found and fail2ban counts it as a single attempt. It going banned if fail2ban would count at least `maxretry` attempts during the `findtime` time window.
In case of `maxretry = 1` the value of `findtime` is not interesting anymore and it causes a ban on first attempt, so this finding message in fail2ban.log should be immediately followed by message `NOTICE  [ip-blacklist] Ban 104.236.0.0/16`, then it is really banned.
I have no idea what exactly will happen with `maxretry = 0`, since it is a bit unexpected configuration. As I already wrote above, you have to change it to `maxretry = 1` to cause immediate ban.

**Comment by Khnaz35** [comment]
@sebres can you tell me how to check these ban ips?
[khnaz35@mail ~]# sudo fail2ban-client status exim
Status for the jail: exim
|- Filter
|  |- Currently failed: 4
|  |- Total failed:     548
|  `- File list:        /var/log/exim/main.log
`- Actions
   |- Currently banned: 0
   |- Total banned:     2
   `- Banned IP list:

**Reply by sebres** [reply]
What do you mean with "to check"?

"Total banned: 2" means that since start of jail 2 IPs get banned, but "Currently banned: 0" means that currently no IPs are banned (the 2 IPs got already unbanned as expired by its `bantime`). You can find them only in fail2ban.log or in its database (unless purged), see https://github.com/fail2ban/fail2ban/issues/3170#issuecomment-984808530
Currently banned can be found either with `status` (as you did) or take a look at `fail2ban-client --help | grep banned`.

**Comment by IvanGlinkin** [comment]
Hey guys,
I have a small related issue to this thread, so if you don't mind ...

```
# head /etc/fail2ban/blocklist/abuseipdb-s100-all.ipv4
1.0.87.120       # JP  AS18144   Enecom,Inc.
1.0.118.87       # JP  AS18144   Enecom,Inc.
1.0.136.77       # TH  AS23969   TOT Public Company Limited
1.0.151.123      # TH  AS23969   TOT Public Company Limited
1.0.152.33       # TH  AS23969   TOT Public Company Limited
1.0.152.106      # TH  AS23969   TOT Public Company Limited
1.0.156.28       # TH  AS23969   TOT Public Company Limited
1.0.157.94       # TH  AS23969   TOT Public Company Limited
1.0.158.159      # TH  AS23969   TOT Public Company Limited
1.0.165.243      # TH  AS23969   TOT Public Company Limited
```

```
# wc /etc/fail2ban/blocklist/abuseipdb-s100-all.ipv4
  807005  5572918 45557119 /etc/fail2ban/blocklist/abuseipdb-s100-all.ipv4
```

```
# cat /etc/fail2ban/filter.d/blocklist.conf 
[Definition]
failregex = ^<HOST>\s+#.*
ignoreregex =
```

```
# cat /etc/fail2ban/jail.local | grep blocklist -A 6
[blocklist]
enabled  = true
port     = anyport
action   = route
filter   = blocklist
logpath  = /etc/fail2ban/blocklist/abuseipdb-s100-all.ipv4
maxretry = 0
findtime = 15552000
bantime  = -1
```

```
# fail2ban-regex /etc/fail2ban/blocklist/abuseipdb-s100-all.ipv4 /etc/fail2ban/filter.d/blocklist.conf

Running tests
=============

Use   failregex filter file : blocklist, basedir: /etc/fail2ban
Use         log file : /etc/fail2ban/blocklist/abuseipdb-s100-all.ipv4
Use         encoding : UTF-8


Results
=======

Failregex: 807005 total
|-  #) [# of hits] regular expression
|   1) [807005] ^<HOST>\s+#.*
`-

Ignoreregex: 0 total

Date template hits:

Lines: 807005 lines, 0 ignored, 807005 matched, 0 missed
[processed in 142.03 sec]
```

```
# fail2ban-client banned
[{'blocklist': []}]
```

is it possible it's not working due to the last ip_list file? Thanks a lot

**Reply by sebres** [reply]
1. Wrong jail backend (you have default `systemd`, so the jail get `systemd` instead of `auto` and will monitor journal instead of logpath)?
Just add `backend = auto` to this jail config (e.g. before/after `logpath`).
2. No timestamp but missing `datepattern = {NONE}`? (or older f2b version)... See also #3861

The common issue with such "lists" is - in sense of fail2ban it is still a log-file, so it'd only recognize the IP that are added to the file after the last known position (which is stored in sqlite database of fail2ban per logpath of jail). So if you modify config/RE, after restart it could recognize the whole file as already processed and will not rescan them.
To cause a rescan you'd need either to simulate the "rotation" (e. g. recreate file, better with different first line to avoid checksum equality check) or to delete it from fail2ban-database or the whole db (stop it, delete its from logs-table or remove the whole db-file, start it). See https://github.com/fail2ban/fail2ban/issues/2909#issuecomment-758036512 for more details.
